controllers/operator.go

```diff
@@ -373,56 +430,62 @@ func (o *operator) estimateDeploy(deploy *appsv1.Deployment) (*appsv1.Deployment
 }
 
 func (o *operator) updateDeploy(deploy *appsv1.Deployment) (*appsv1.Deployment, error) {
-	partition, err := helpers.ParsePartitionAnnotation(deploy.Annotations[PartitionAnnotation])
+	consumerId, err := helpers.ParseIntAnnotation(deploy.Annotations[ConsumerAnnotation])
 	if err != nil {
 		return nil, err
 	}
+	partitions := o.assignments[consumerId]
 	deploy.Annotations[GenerationAnnotation] = o.observedGeneration()
 	deploy.Spec = o.consumer.Spec.DeploymentTemplate
 	for i := range deploy.Spec.Template.Spec.Containers {
 		var resources *corev1.ResourceRequirements
 		var underProvision int64
 		container := &deploy.Spec.Template.Spec.Containers[i]
 		if container.Resources.Requests.Cpu().IsZero() {
-			resources, underProvision = o.allocateResources(container, partition)
+			resources, underProvision = o.allocateResources(container, partitions)
 		} else {
-			resources, underProvision = o.updateResources(container, partition)
+			resources, underProvision = o.updateResources(container, partitions)
 		}
 		o.updateScaleAnnotations(deploy, underProvision)
 		container.Resources = *resources
-		container.Env = helpers.PopulateEnv(container.Env, &container.Resources, o.consumer.Spec.PartitionEnvKey, int(partition))
+		container.Env = helpers.PopulateEnv(container.Env, &container.Resources, o.consumer.Spec.PartitionEnvKey, partitions)
 	}
 	return deploy, nil
 }
 
-func (o *operator) constructDeploy(partition int32) *appsv1.Deployment {
-	deployLabels := make(map[string]string)
+func (o *operator) constructDeploy(consumerId int32) *appsv1.Deployment {
+	partitionIds := o.assignments[consumerId]
+	deployLabels := map[string]string{
+		"app":        o.consumer.Spec.Name,
+		"controller": o.consumer.Name,
+	}
 	deployAnnotations := make(map[string]string)
 	deploy := &appsv1.Deployment{
 		TypeMeta: metav1.TypeMeta{},
 		ObjectMeta: metav1.ObjectMeta{
 			Labels:      deployLabels,
 			Annotations: deployAnnotations,
-			Name:        fmt.Sprintf("%s-%d", o.consumer.Spec.Name, partition),
+			Name:        fmt.Sprintf("%s-%s", o.consumer.Spec.Name, strings.Join(helpers.Int2Str(partitionIds), "-")),
```

Comment:
nit: would be nice partitions to be sorted first

Reply:
same here, array is created sorted